### PR TITLE
VY-222 Refactor code in employee form tab.

### PR DESCRIPTION
**Pull request by lf-kiranpariyar**
@lf-romitamgai . Please review the code.

### CODE REVIEW

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

File: frontend/src/js/reducers/crudReducer.js [hunk 1]
```diff
@@ -83,14 +83,14 @@ let crudReducer = function (state = initialState, action) {
...
       newState = _.cloneDeep(state);
-      newState.imageURL = action.value;
+      newState.imageData = action.value;
       return newState;
 
-    case 'CLEAR_IMAGE_ITEM':
+    case actionTypeConstants.CLEAR_IMAGE_DATA:
       newState = _.cloneDeep(state);
-      newState.imageURL = '';
+      newState.imageData = '';
```
Comment (lf-romitamgai) [inline frontend/src/js/reducers/crudReducer.js]: imageData should initially be an empty object

File: frontend/test/reducers/crudReducer.spec.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ const initialState = {
     imageFile: {}
   },
   pagination: {},
-  imageURL: ''
+  imageData: ''
```
Comment (lf-romitamgai) [inline frontend/test/reducers/crudReducer.spec.js]: same here

File: frontend/src/js/components/employee/employee-form/tabbed-components/GeneralInfo.js [hunk 3]
```diff
@@ -263,7 +267,7 @@ class PersonalInfo extends React.Component {
 
         <ImagePreview
           isValidImage={this.props.formValidator.validateImage}
-          initialImageURL={this.props.imageURL}
+          initialImageURL={this.props.imageData}
```
Comment (lf-romitamgai) [inline frontend/src/js/components/employee/employee-form/tabbed-components/GeneralInfo.js]: is it still imageURL?

File: frontend/src/js/actions/crudActions.js [hunk 4]
```diff
@@ -281,21 +281,18 @@ let crudActions = {
     }
   },
 
-  updateImageFile(entity,value) {
+  updateImageData(value) {
     return{
-      type: 'UPDATE_IMAGE_URL',
```
Comment (lf-romitamgai) [inline frontend/src/js/actions/crudActions.js]: change here too

File: frontend/src/js/actions/crudActions.js [hunk 5]
```diff
@@ -209,18 +209,18 @@ let crudActions = {
     }
   },
 
-  uploadImage(entity, file, id, callback){
+  uploadImage(entity, file, id, callback, addEmployee){
```
Comment (FriedDust) [inline frontend/src/js/actions/crudActions.js]: why isn't callback the last parameter?

File: frontend/src/js/actions/crudActions.js [hunk 6]
```diff
@@ -209,18 +209,18 @@ let crudActions = {
...
       var oldRoute = getState().routing.locationBeforeTransitions.pathname;
       dispatch(apiActions.apiRequest());
       return (coreApiService.uploadPhoto(entity, file, id)
         .then(()=> {
           if (actionService.isSameRoute(getState, oldRoute)) {
             dispatch(apiActions.apiResponse());
-            if(id) {
-              Toastr.success(messageConstants.SUCCESSFULLY_UPDATED_EMPLOYEE);
-            }else{
+            if(addEmployee) {
```
Comment (FriedDust) [inline frontend/src/js/actions/crudActions.js]: code format

**Pull request merged**